frontend/src/Down.tsx

```diff
@@ -0,0 +1,39 @@
+import { TwitterButton } from './components/TwitterButton';
+import { GLOBALS } from './index';
+
+export const Down = () => {
+    // Every 30 seconds, pull the API and see if it's back up
```

Comment:
setInterval will not stop when the component gets unmounted, i.e when the site goes non-offline it still keeps fetching in the background.

I don't recall the react methods for running something when a component is unmounted but `setInterval` at least returns the id of the interval once run which can be used with `clearInterval(ID)` for canceling the interval.

Reply:
Will try to implement.

Reply:
It does location.refresh() (simplest approach I could think of), so it never gets unmounted. How is this still a problem?